apis/voyager/v1beta1/validator.go

```diff
@@ -274,6 +296,31 @@ func checkOptionalPort(port intstr.IntOrString) (int, error) {
 	return 0, fmt.Errorf("invalid data type %v for port %s", port.Type, port)
 }
 
+func checkOptionalAddress(address string) (string, error) {
+	if address != "" && net.ParseIP(address) == nil {
+		return "", fmt.Errorf("could not parse IPv4 or IPv6 address")
+	} else if address == "" {
+		return "*", nil
+	}
+
+	return address, nil
+}
+
+func checkExclusiveWildcard(address string, port int, defined map[string]*address) error {
+	var wildcard = fmt.Sprintf("*:%d", port)
+	if address == "*" {
+		if _, ok := defined[wildcard]; !ok && len(defined) > 0 {
```

Comment:
This should check for entries that have `<non-*-addr>:port` defined.

Reply:
Apologies if I'm misunderstanding the comment, but this is what this code does (unless, of course, I've botched it). Specifically, there's two cases:

If we're attempting to register a wildcard address, check whether there's an existing wildcard address registered for the same port. If not, check whether there's any other rules registered. If there are, we can assume they're specific addresses, and thus fail out.

If we're attempting to register a non-wildcard address, check whether there's an already-existing wildcard address. If there is, fail out.

Pretty sure this covers everything, right?

Reply:
` If not, check whether there's any other rules registered. If there are, we can assume they're specific addresses, and thus fail out.`

This defined map will have rules for all other ports, too. So, I think you need to check the there are other rules for the same port but with specific bind address, right?

Reply:
Ah, correct, my mistake. I'll fix.